dags/hindawi/utils.py

```diff
@@ -0,0 +1,48 @@
+import uuid
+from datetime import date, datetime
+from io import BytesIO
+from xml.etree import ElementTree
+
+from airflow.api.common import trigger_dag
+
+
+def save_file_in_s3(data, repo):
+    if data:
+        date_today = date.today().strftime("%Y-%m-%d")
+        prefix = datetime.now().strftime("%Y-%m-%dT%H:%M")
+        key = f"{date_today}/{prefix}.xml"
+        byte_file = BytesIO(data)
+        repo.save(key, byte_file)
+        return key
+
+
+def split_xmls(repo, key):
+    ids_and_records = []
+    file = repo.find_by_id(key)
+    xml_string = file.getvalue().decode("utf-8")
+    for tag in ElementTree.fromstring(xml_string):
+        if "ListRecords" not in tag.tag:
+            continue
+        for record in tag:
+            if "record" not in record.tag:
+                continue
+            id = str(uuid.uuid4())
+            ids_and_records.append(
+                {
+                    "id": id,
+                    "record": ElementTree.tostring(record).decode("utf-8"),
+                }
+            )
+    return ids_and_records
+
+
+def trigger_file_processing_DAG(ids_and_records):
+    if len(ids_and_records) > 0:
```

Comment:
```suggestion
    if not ids_and_records:
       // Log that no ids or records to trigger the dag
       return
```

It will keep the loop out of the if and one less indent.

Reply:
we are already logging that there are no articles uploaded, I don't know is worth it to log it again